### DISCUSSION TITLE
Text-search gff file

### DISCUSSION THREAD

**Discussion by tbrown91** [GMOD/jbrowse-components#2786] [Q&A] [answered]
Hi,

I am able to search my gff file using the jbrowse2 online version by following the steps online (bgzip, tabix & text-index) but this is not working for me with the desktop version. I can see that this works if I download reference datasets so feel like I'm missing something.

Is there a way to load the .ix or .ixx files into jbrowse2 desktop - or can I create a Jbrowse1TextSearchAdapter-associated index?

At the moment I am loading gff and gff.csi in as Gff3TabixAdapter and under the textSearchAdapter settings it has chosen Jbrowse1TextSearchAdapter, with no file chosen. If I try to change this to trixTextSearchAdapter the drop-down menu closes and doesn't change the selection.

Any help would be much appreciated,

Tom

**Comment by tbrown91** [comment]
Hi again,

A small follow-up to say that the search function seems to be working pretty well within jbrowse1 desktop version. Here I searched for "XRN1" and show the auto-complete:

![image](https://user-images.githubusercontent.com/11500978/156362852-7ecb8fd1-1bf9-418f-9359-645e8e71bbf1.png)

Doing the same thing within jbrowse2 doesn't auto-complete. I'd like to be able to view the gff and HiC data with searching possible, so some way to combine all of this would be amazing.

Tom

**Comment by cmdcolin** [comment, marked as answer]
@teresam856 is working on text indexing for jbrowse desktop at the moment

if you want to try to use something in the interim, you can use the techniques of indexing for the web on desktop, you find the path of your config.json file of interest (listed in table on the start screen for example) and run

`jbrowse text-index --out /path/to/specific/config.json`

this will create a text index for that config

**Reply by cmdcolin** [reply]
to be clear, with @teresam856 's work, we will have a built in way to make this easier :)

**Reply by tbrown91** [reply]
Hi Colin, 

That's what I have been trying to do. At the moment I have the following three files:

trix/asm.ix
trix/asm.ixx
tris/asm_meta.json

but I am not able to load these into jbrowse. When I go to the track settings and try to select "TrixTextSearchAdapter" this option fails and keeps me with JBrowse1TextSearchAdapter selected:

![image](https://user-images.githubusercontent.com/11500978/156525894-2d813b55-6c70-423d-8245-a035aa9b9a2d.png)

Have I selected the wrong thing or am trying to load the wrong file?

Tom

**Reply by cmdcolin** [reply]
I am testing out this workflow myself (running text-index on a jbrowse-desktop created file) and see there may be some issues. It shouldn't require any configuration editing but I ran into a couple things

1) The text-index command outputs 'uri' but the desktop expects 'localPath' for files on the local machine. You could fix this be manually editing the 'uri': 'trix/asm.ix' to 'localPath':'/path/to/file/on/machine/asm.ix'  (i did this by manually editing the json)
2) The files outputted by text-index may be empty in some cases (ran into this in my testing)

I would suggest using jbrowse-web for text indexing for the moment but we will be trying to add this to desktop soon.

**Reply by tbrown91** [reply]
Hi Colin,

Thank you for the help. We got there eventually. Just to make it super clear - I had to save my current session and then edit the .jbrowse file with the adapter file paths and formats before re-opening with jbrowse2. I don't know why this option to load the .ix, .ixx and meta.json weren't working within the jbrowse GUI, but in the background this seems to be working fine.

Tom
